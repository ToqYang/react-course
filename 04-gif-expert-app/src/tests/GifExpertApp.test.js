import React from "react";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Test component <GifExpertApp />", () => {
  test("Test correct making component", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe demostrar lista de categorias ", () => {
    const categories = ["One Punch", "Dragon Ball"];
    const wrapper = shallow(<GifExpertApp categories={categories} />);
    expect(wrapper.find("GifGrid").length).toMatchSnapshot(categories.length);
  });
});
