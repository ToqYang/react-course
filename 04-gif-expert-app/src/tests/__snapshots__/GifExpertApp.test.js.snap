// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component <GifExpertApp /> Debe demostrar lista de categorias  1`] = `0`;

exports[`Test component <GifExpertApp /> Test correct making component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
